topostats/processing.py

```diff
@@ -317,7 +364,12 @@ def process_scan(
         LOGGER.info(f"[{filename}] Detection of grains disabled, returning empty data frame.")
         results = create_empty_dataframe()
 
-    return image_path, results
+    # Set the index to zero since we are only adding one row. Will crash otherwise since we are not
+    # passing iterables.
+    image_stats_df = pd.DataFrame(image_stats, index=[0])
+    image_stats_df.set_index("filename", inplace=True)
```

Comment:
If the below suggestion of making a dictionary of results and `pd.concat()`ing that then this line would no longer be needed.